File: src/my_project/easy_problems/from151To200/special_array_with_x_elements_greater.py
from typing import List 

class Solution:
    def specialArray(self, nums: List[int]) -> int:

        nums.sort(reverse=True)
        print(nums)
        len_nums: int = len(nums)

        for i in range(len_nums):
            if i+1 <= nums[i]:

                if i == len_nums -1:
                    return i +1 
                else: 
                    if i + 1 <= nums[i+1]:
                        continue
                    else: 
                        return i+1

        return -1


solution = Solution()


print(solution.specialArray(nums = [3,5]))

print(solution.specialArray(nums = [0,0]))

print(solution.specialArray(nums = [0,4,3,0,4]))

print(solution.specialArray(nums = [3,6,7,7,0]))

File: src/my_project_typescript/easy_problems_typescript/from51To100/specialArrayWithXElementsGreater.ts
function specialArray(nums: number[]): number {


    nums.sort( (a,b) => a>=b?-1:1)

    let lenNums: number = nums.length

    for (let i =0; i< nums.length; i++){
        if (i+1 <= nums[i]){
            if (i == nums.length -1){
                return i+1
            }
            else{
                if (i+1 <= nums[i+1]){
                    continue
                }
                else{
                    return i+1
                }
            }
        }
    }
    return -1
};


console.log(specialArray([3,5]))

console.log(specialArray([0,0]))

console.log(specialArray([0,4,3,0,4]))

console.log(specialArray([3,6,7,7,0]))
